airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -463,6 +463,9 @@ def wait_for_job(self, url: str) -> str:
                     raise AirbyteTracedException(message=message, failure_type=FailureType.config_error, exception=error)
                 else:
                     raise error
+            job_id = job_info["id"]
```

Comment:
I'm fine with this change. However, this will lead some tests to fail because the [HTTP response here](https://github.com/airbytehq/airbyte/blob/3da2fd34219726c95a8a7bdb7c8e293f25fb214d/airbyte-integrations/connectors/source-salesforce/unit_tests/integration/test_bulk_stream.py#L70-L73) is not representative of the actual response defined [here](https://developer.salesforce.com/docs/atlas.en-us.248.0.api_asynch.meta/api_asynch/asynch_api_reference_jobinfo.htm). I'm not sure if other tests might fail as well. 

If you are all-in to make the response configurable, you can explore how to create a builder that would allow us to configure the response. Something like:

```
class SalesforceJobResponseBuilder:
    def __init__(
        self,
        records_path: Union[FieldPath, NestedPath],
        pagination_strategy: Optional[PaginationStrategy]
    ):
        self._response = json.dumps(find_template("job_response", __file__))
        self._status_code = 200

    def with_id(self, id: str) -> "HttpResponseBuilder":
        self._response["id"] = id
        return self

    def with_state(self, state: str) -> "HttpResponseBuilder":
        self._response["state"] = state
        return self

    def with_status_code(self, status_code: int) -> "HttpResponseBuilder":
        self._status_code = status_code
        return self

    def build(self) -> HttpResponse:
        return HttpResponse(json.dumps(self._response), self._status_code)
```

We can sync on that tomorrow if you'd like

Reply:
done!